internal/provider/resource_storagezone.go

```diff
@@ -256,14 +251,6 @@ func resourceStorageZoneUpdate(ctx context.Context, d *schema.ResourceData, meta
 
 	updateErr := clt.StorageZone.Update(ctx, id, storageZone)
 	if updateErr != nil {
-		// if our update failed then revert our values to their original
-		// state so that we can run an apply again.
-		revertErr := revertUpdateValues(d)
```

Comment:
@fho this is actually currently an important bit, the reason is that there are certain values that never get read out of bunny state on a `get` (specifically the ones referenced in `revertUpdateValues`). Additionally, there is validation of these properties that happens on Bunny's side which can fail. If that validation fails on an apply we need to manually revert those values back to what they were before otherwise TF gets confused.

Thats not super clear, so here's a concrete example:

1. Add ` custom_404_file_path  = "/foo.html"` to your resource
2. Run an `apply` (this will succeed and correctly modify that value)
3. Update ` custom_404_file_path  = "/bar"` on your resource
4. Run an `apply` this apply will fail due to a validation error in Bunny
5. Run a `plan` immediately after the failed `apply` and see that TF now thinks that the state is up-to-date even though it's incorrect as the path is actually `/foo.html` and not `/bar`.


To my knowledge, this is an artifact of how those properties can't be read out of Bunny state so we need to infer their state based on successful or errored responses on update.

Hopefully that makes sense. If there's a better way to handle this I'm all ears.